libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/LocaleUtils.java

```diff
@@ -52,13 +61,24 @@ public static Locale getLocale(Context context) {
   public static void setLocale(Context context, Locale locale, @NavigationUnitType.UnitType int unitType) {
     SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(context).edit();
 
+    setLocale(editor, locale, unitType);
+
+    editor.apply();
+  }
+
+  /**
+   * Sets locale and unitType into SharedPreferences for later use
+   * @param editor to add locale and unitType to
+   * @param locale to set language
+   * @param unitType to set unitType to use
+   */
+  public static void setLocale(
+    SharedPreferences.Editor editor, Locale locale, @NavigationUnitType.UnitType int unitType) {
     editor.putInt(NavigationConstants.NAVIGATION_VIEW_UNIT_TYPE, unitType);
     if (locale != null) {
       editor.putString(NavigationConstants.NAVIGATION_VIEW_LOCALE_LANGUAGE, locale.getLanguage());
       editor.putString(NavigationConstants.NAVIGATION_VIEW_LOCALE_COUNTRY, locale.getCountry());
     }
-
-    editor.apply();
```

Comment:
The javadoc explains this method sets the locale / unit type to preferences, but it looks like we aren't applying these changes